vermouth/tests/test_selectors.py

```diff
@@ -0,0 +1,148 @@
+# Copyright 2018 University of Groningen
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+Tests for the selectors in :mod:`vermouth.selectors`.
+"""
+
+import functools
+
+import pytest
+
+import vermouth
+from vermouth.pdb.pdb import read_pdb
+from vermouth.tests.datafiles import (
+    PDB_PROTEIN,
+    PDB_NOT_PROTEIN,
+    PDB_PARTIALLY_PROTEIN,
+)
+
+
+@pytest.mark.parametrize(
+    'molecule, reference_answer',
+    [(read_pdb(str(path)), answer) for path, answer in [
+        (PDB_PROTEIN, True),
+        (PDB_NOT_PROTEIN, False),
+        (PDB_PARTIALLY_PROTEIN, False),
+    ]]
+)
+def test_is_protein(molecule, reference_answer):
+    """
+    Make sure that proteins are correctly identified as such.
+    """
+    assert vermouth.selectors.is_protein(molecule) == reference_answer
+
+
+@pytest.mark.parametrize(
+    'atom, reference_answer',
+    (
+        ({'position': [0, 0, 0]}, True),
+        ({'position': None}, False),
+        ({}, False),
+    )
+)
+def test_selector_no_position(atom, reference_answer):
+    """
+    Test that :func:`vermouth.selectors.selector_has_position` works as expected.
+    """
+    assert vermouth.selectors.selector_has_position(atom) == reference_answer
+
+
+@pytest.mark.parametrize('node', (
+    {},
+    {'resname': 'PLOP', 'atomname': 'A'},
+))
+def test_select_all(node):
+    """
+    Test that :func:`vermouth.selectors.select_all` indeed accept every nodes.
+    """
+    assert vermouth.selectors.select_all(node)
+
+
+@pytest.mark.parametrize('node, expected', (
+    ({}, False),  # empty node
+    ({'atomname': 'not BB'}, False),  # wrong atomname
+    ({'resname': 'anything'}, False),  # no atomname
+    ({'atomname': 'BB'}, True),  # only correct atomname
+    ({'atomname': 'BB', 'other': 'something'}, True),  # correct atomname + other attribute
+))
+def test_select_backbone(node, expected):
+    """
+    Test that :func:`vermouth.selectors.select_backbone` only select beads with
+    'BB' as atom name.
+    """
+    assert vermouth.selectors.select_backbone(node) == expected
+
+
+@pytest.mark.parametrize('node, attribute, values, expected', (
+    ({}, 'name', ['something', 'other'], False),
+    ({'name': 'nothing'}, 'name', ['something', 'other'], False),
+    ({'name': 'something'}, 'name', ['something', 'other'], True),
+    ({'name': 'nothing', 'plop': 'other'}, 'name', ['something', 'other'], False),
+    ({'name': 'nothing'}, 'name', [], False),
```

Comment:
`'name': 'other'`
`'name': 'somethingother'`

Reply:
What are these testing?

Reply:
`({'name': 'other'}, 'name', ['something', 'other'], False),` To make sure any of the allowed names works.
`({'name': 'somethingother'}, 'name', ['something', 'other'], False),` To make sure the equality isn't actually doing membership testing